File: set_1/arrays/python/9a61c304-d6f7-40cb-a057-1aa93c25c865.py
# Solution 1:
def reverse_array(arr, start, end):
	while (start < end):
		temp = arr[start]
		arr[start] = arr[end]
		arr[end] = temp
		start += 1
		end = end-1
	return arr

def left_rotate(arr, pivot):

	if pivot == 0:
		return
	arr_length = len(arr)
	pivot = pivot % arr_length
	result_array = reverse_array(arr, 0, pivot-1)
	result_array = reverse_array(arr, pivot, arr_length-1)
	result_array = reverse_array(arr, 0, arr_length-1)
	return result_array


arr1 = [24, 26, 30, 46, 59, 60, 61, 64, 67, 69]
pivot_index = 2
print(f"Original input array --> {arr1}")
res1 = left_rotate(arr1, pivot_index)
print(f"Array after rotating --> {res1}")

File: set_1/arrays/javascript/9a61c304-d6f7-40cb-a057-1aa93c25c865.js
// Solution 1:
const reverseArray = (arr, start, end) => {
    while (start < end) {
        var temp = arr[start];
        arr[start] = arr[end];
        arr[end] = temp;
        start++;
        end--;
    }
    return arr
}

const leftRotate = (arr, pivot) => {
    const arrLength = arr.length;
    if (pivot == 0) return;
    pivot = pivot % arrLength;

    let resultArray = reverseArray(arr, 0, pivot - 1);
    resultArray = reverseArray(arr, pivot, arrLength - 1);
    resultArray = reverseArray(arr, 0, arrLength - 1);
    return resultArray
}



const arr1 = [13, 15, 17, 22, 25, 32, 37, 39, 41, 42];
const pivotIndex = 2;
console.log(`Original input array --> [${arr1}]`)
const res1 = leftRotate(arr1, pivotIndex);
console.log(`Array after rotating --> [${res1}]`)
